### PR TITLE
Added `log_all_trials` param to `NeptuneCallback`

### PR DESCRIPTION
**Pull request by SiddhantSadangi**
Added a `log_all_trials` param to `NeptuneCallback` to allow disabling logging of all trials, similar to what `log_all_trials` does in `log_study_metadata()`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## neptune-optuna 1.1.1
```
Comment (Raalsky) [inline CHANGELOG.md:1]: `1.2.0` then according to Semversion, right?

File: pyproject.toml [hunk 2]
```diff
@@ -25,6 +25,7 @@ pytest = { version = ">=5.0", optional = true }
 pytest-cov = { version = "2.10.1", optional = true }
 deepdiff = { version = ">=6.2.3", optional = true }
 neptune = { version = ">=1.0.0", optional = true }
+cffi = { version = "*", optional = true}
```
Comment (Raalsky) [inline pyproject.toml:28]: You have to add i under `dev` extras (line 37) if you want to use `pip install -e .[dev]`.

File: src/neptune_optuna/impl/__init__.py [hunk 3]
```diff
@@ -165,6 +167,7 @@ def __init__(
             (bool, type(None)),
         )
         verify_type("target_names", target_names, (list, type(None)))
+        verify_type("log_all_trials", log_all_trials, (bool, type(None)))
```
Comment (Raalsky) [inline src/neptune_optuna/impl/__init__.py:170]: We cannot provide `None` if I understand :wink:
```suggestion
        verify_type("log_all_trials", log_all_trials, bool)
```

**Comment by SiddhantSadangi** [comment]
@Raalsky - when you have a while, could you check the failure with `Deepdiff`? I am not able to reproduce it locally on Windows

**Comment by Raalsky** [comment]
@SiddhantSadangi Try to pin `deepdiff` to `deepdiff==6.6.1` (or just `<6.7`).

**Comment by SiddhantSadangi** [comment]
> @SiddhantSadangi Try to pin `deepdiff` to `deepdiff==6.6.1` (or just `<6.7`).

@Raalsky - no luck

**Comment by Raalsky** [comment]
Let's .... skip the test :wink:. You should be able to use something like `pytest.mark.skip` + please post something to Jira about it :wink:

**Comment by SiddhantSadangi** [comment]
@Raalsky - fixed 🎉

**Review by Raalsky — APPROVED** [review]

**Comment by SiddhantSadangi** [comment]
@Raalsky - keeping the branch in case there's further feedback from the user before we publish a stable release

**Pull request merged**